app/controllers/bank_accounts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BankAccountsController < ApplicationController
+  def index
+    if current_user.banker ==true || current_user.staff == true
+      @accounts = BankAccount.all
+      characters = Character.all
+      @characters_without_accounts = []
+      characters.each do |character|
+        if !BankAccount.has_account?(character.id) == true && character.archived==false
+          @characters_without_accounts << character
+        end
+      end
+      @account = BankAccount.new
+    else
+      authenticate_staff!
```

Comment:
Weird to do this in an `else` flow, instead of as a `before_action`. Come to mention it, does this method even exist?

Reply:
authenticate_staff! is a method you wrote, which kicks the user back to the home page if they aren't staff, so this seemed dryer than writing out another redirect/flash message.

Reply:
What you want to do is add `before_action :authenticate_staff!` as basically the second line of this controller (as done in, for example, the `Staff::BookingsController`. This makes that authentication step happen before anything else, so you don't have to worry about it.